discovery-provider/src/tasks/index.py

```diff
@@ -243,6 +248,101 @@ def index_blocks(self, db, blocks_list):
     if num_blocks > 0:
         logger.warning(f"index.py | index_blocks | Indexed {num_blocks} blocks")
 
+
+def get_user_track_play_counts(session, user_id, track_id):
+    query = text(
+        f"""
+        select play_item_id, count
+        from aggregate_plays
+        where play_item_id in :ids
+        """
+    )
+    query = query.bindparams(bindparam('ids', expanding=True))
+
+    track_play_counts = session.execute(query, { "ids": track_ids }).fetchall()
+    track_play_dict = dict(track_play_counts)
+    return track_play_dict
+
+# Retrieve the play counts from the identity service
+# NOTE: indexing the plays will eventually be a part of `index_blocks`
+def get_track_plays(self, db):
```

Comment:
can this go in its own file like the new queries have been?

Reply:
this isn't really a queries thing though, it's very specific to the indexing flow. I think it should go here, but mabye in a separate file as a helper. Are you ok w/ that @hareeshnagaraj ?

Reply:
I put this, the db query and indexing in a separate file b/c each of the celery jobs have their own file and I don't think any of the logic would be reused outside of the job.